doc/tutorials/developing-with-dune/introduction.md

```diff
@@ -0,0 +1,169 @@
+Introduction
+============
+
+The goal of this first chapter is to get to a place where you have a working
+Dune project with a skeleton of a calculator.
+
+This is is a [tutorial](https://diataxis.fr/tutorials/): it is meant to be
+followed in order.
+
+## Installing Packages
+
+First, you'll need to have a working Opam installation. This is described in
+{doc}`/howto/install-dune`.
+
+Then, create an empty directory somewhere, say `~/dune-calc`. In this tutorial,
+we will only create files in this directory.
+
+Let's first make sure you have a working opam installation. Run this command:
+
+```sh
+opam --version
+```
+
+It should display something like "2.2.0". Anything greater than 2.0.0 is fine.
+
+:::{important}
+When asked to type a command, you can click {octicon}`copy` to copy the full command.
+
+In this tutorial, all commands will be typed at the root of your project, like
+`~/dune-calc`.
+:::
+
+Let's create a local switch: `cd` to this directory and run the following command.
+This can take a few minutes.
+
+```sh
+opam switch create ./ 5.1.1
+```
+
+This command has created a directory named `_opam` in the current directory.
+Now, let's install some packages by running:
+
+```sh
+opam install dune.3.15.3 menhir.20231231
+```
+
+You can confirm that `opam` is correctly setup by typing `dune --version`,
+which should display 3.15.3. Otherwise, please refer to
+{doc}`/howto/install-dune`.
+
+## The Calculator Skeleton
+
+Now that we have an opam switch and some packages installed, let's create the
+various files.
+
+For each file, click {octicon}`chevron-down` to reveal the file contents and
+click {octicon}`copy` to copy the contents to your clipboard. Open `file.txt`
+in a text editor and paste the contents there.
+
+::::{dropdown} `dune-project`
+:icon: file-code
+
+:::{literalinclude} introduction/dune-project
+:language: dune
+:::
+
+This file contains metadata about the project:
+
+- the version of the dune language we're using
+- the extensions we're {doc}`using </reference/dune-project/using>`
+- the {doc}`package </reference/dune-project/package>` we're defining
+
+:::{seealso}
+{doc}`/reference/dune-project/index`
+  Reference documentation about `dune-project` files
+:::
+::::
+
+::::{dropdown} `dune`
+:icon: file-code
+
+:::{literalinclude} introduction/dune
+:language: dune
+:::
+
+This file contains a description of what's in our project:
+
+- an {doc}`/reference/dune/executable` stanza defining our calculator binary
+- an {doc}`/reference/dune/ocamllex` stanza, setting up rules to compile
```

Comment:
I think that's the scope of another tutorial. I want to create the actual "first steps with dune" tutorial which would be started from nothing and cover the basics of dune (the files look like this, it rebuilds your project when you change the files, etc).

But I also agree that frontloading menhir is a bit too much, since it's a missed opportunity to explain `(menhir)` and `(using)`. So maybe we could start with an even simpler calculator that just splits on `+` and parses the rest as integers; and in a first step we add menhir.
Would that work?

Reply:
Yes, I think if you add it as an explicit step in the tutorial ("replace hand-written 'parser' with Menhir") I still think it would be an improvement.